md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/AnalyticsAppAPI.java

```diff
@@ -0,0 +1,281 @@
+        }
+        return restWrapper;
+    }
+
+    @RequestMapping(value = {"/apps/{industry}/{category}"}, method = RequestMethod.GET)
+    @ResponseBody
+    public RestWrapper listApps(@PathVariable("industry") String industry,@PathVariable("category") String category, Principal principal) {
+
+        RestWrapper restWrapper = null;
+        try {
+            LOGGER.info("Categpry is "+category);
+            LOGGER.info("industry is "+industry);
+            List<AnalyticsApps> analyticsAppsList = new ArrayList<AnalyticsApps>();
+            for (String app : analyticsAppsDAO.listApps(industry, category)) {
+                AnalyticsApps analyticsApp = new AnalyticsApps();
+                analyticsApp.setIndustryName(industry);
+                analyticsApp.setCategoryName(category);
+                analyticsApp.setAppName(app);
+                analyticsAppsList.add(analyticsApp);
+            }
+
+            restWrapper = new RestWrapper(analyticsAppsList, RestWrapper.OK);
+            LOGGER.info("All records listed from Properties by User:"+category+industry );
+
+        } catch (MetadataException e) {
+            LOGGER.error(e);
+            restWrapper = new RestWrapper(e.getMessage(), RestWrapper.ERROR);
+        }
+        return restWrapper;
+    }
+
+    @RequestMapping(value = {"/dashboardurl/{industry}/{category}/{app}"}, method = RequestMethod.GET)
+    @ResponseBody
+    public RestWrapper getDashBoardURL(@PathVariable("industry") String industry,@PathVariable("category") String category,@PathVariable("app") String app, Principal principal) {
+
+        RestWrapper restWrapper = null;
+        try {
+            LOGGER.info("app is "+app);
+            LOGGER.info("Categpry is "+category);
+            LOGGER.info("industry is "+industry);
+            List<AnalyticsApps> analyticsAppsList = new ArrayList<AnalyticsApps>();
+            for (String durl : analyticsAppsDAO.getDashBoardURL(industry, category, app)) {
+                AnalyticsApps analyticsApp = new AnalyticsApps();
+                analyticsApp.setIndustryName(industry);
+                analyticsApp.setCategoryName(category);
+                analyticsApp.setAppName(app);
+                analyticsApp.setDashboardUrl(durl);
+                analyticsAppsList.add(analyticsApp);
+            }
+
+            restWrapper = new RestWrapper(analyticsAppsList, RestWrapper.OK);
+            LOGGER.info("All records listed from Properties by User:");
+
+        } catch (MetadataException e) {
+            LOGGER.error(e);
+            restWrapper = new RestWrapper(e.getMessage(), RestWrapper.ERROR);
+        }
+        return restWrapper;
+    }
+
+    @RequestMapping(value = {"/json/{industry}/{category}/{app}"}, method = RequestMethod.GET)
+    @ResponseBody
+    public RestWrapper getJson(@PathVariable("industry") String industry,@PathVariable("category") String category,@PathVariable("app") String app, Principal principal) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused method parameter(s) "principal". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1172)